varcode/variant_collection.py

```diff
@@ -110,49 +67,55 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    def clone(self, new_variants=None):
+    def clone_metadata(self, new_variants):
         """
         Create copy of VariantCollection with same metadata but possibly
-        different Variant entries. If no variants provided, then just make a
-        copy of self.variants.
+        different Variant entries.
         """
-        if new_variants is None:
-            new_variants = self.variants
+
         return VariantCollection(
-            variants=list(new_variants),
-            original_filename=self.original_filename,
-            reference_path=self.reference_path,
-            reference_name=self.reference_name,
-            ensembl_release=self.ensembl_release)
+            variants=new_variants,
+            original_filename=self.original_filename)
+
+    def clone(self):
+        return self.clone_metadata(list(self.variants))
 
     def drop_duplicates(self):
         """
         Create a new VariantCollection without any duplicate variants.
         """
-        return self.clone(set(self.variants))
+        return self.clone_metadata(set(self.variants))
```

Comment:
I think there's a bug floating around here, if we're now returning an instance that has a `set` of variants instead of a `list`, since the `__eq__` method checks equality by iterating through the zip'd variants, which assumes a particular sort order. I'm confused at why the `__eq__` method does that actually. My suggestion is to (1) drop the `__eq__` method -- I think users will just want to compare the .variants using usual python semantics, instead of depending on the intrinsically shaky semantics of equality in VariantCollection (e.g. what if the filename is different, is that a different collection? I just looked at that code an have already forgotten) and (2) either remove this `drop_duplicates` method, or modify it to create a VariantCollection with a `list` of variants instead of a set (this will require removing duplicates in a way that preserves sort order)